File: example/claude/py/subagent/agent.py
import asyncio
from claude_agent_sdk import query, ClaudeAgentOptions


async def main():
    async for message in query(
            prompt="Analyze this codebase for security vulnerabilities",
            options=ClaudeAgentOptions(allowed_tools=["Read", "Glob", "Grep", "Task"])
    ):
        print(message)


asyncio.run(main())

File: example/claude/ts/subagent/agent.ts
import { query } from "@anthropic-ai/claude-agent-sdk";

// Using top-level await (ESM only)
// @ts-ignore
for await (const message of query({
    prompt: "Analyze this codebase for security vulnerabilities",
    options: {
        allowedTools: ["Read", "Glob", "Grep", "Task"]
    }
})) {
    console.log(message);
}